src/App.tsx

```diff
@@ -1,29 +1,29 @@
-import { Component } from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 
-interface State {
-  movies: Movie[];
-}
+export const App: React.FC = () => {
+  const [movies, addMovie] = useState<Movie[]>([]);
 
-export class App extends Component<{}, State> {
-  state: State = {
-    movies: [],
-  };
+  const onAddMovie = useCallback((newMovie: Movie): void => {
+    const moviesIds = movies.map(
+      ({ imdbID }) => imdbID,
+    );
```

Comment:
why do you recreating this array to every method call? just move it component and cover with useMemo